### PR TITLE
change to reference values

### PR DESCRIPTION
**Pull request by Laurynas-Jagutis**
Related to https://github.com/PowerGridModel/power-grid-model/issues/267

Fixes performance-unnecessary-copy-initialization code smells.
Changed local variable declarations to a const reference where applicable.

bugprone-narrowing-conversions did not display any code smells.

### CODE REVIEW

File: tests/cpp_validation_tests/test_validation.cpp [hunk 1]
```diff
@@ -443,7 +443,7 @@ std::vector<CaseParam> read_all_cases(bool is_batch) {
     for (std::string calculation_type : {"power_flow", "state_estimation", "short_circuit"}) {
         // loop all sub-directories
         for (auto const& dir_entry : std::filesystem::recursive_directory_iterator(data_path / calculation_type)) {
-            std::filesystem::path const case_dir = dir_entry.path();
+            const std::filesystem::path& case_dir = dir_entry.path();
```
Comment (mgovers) [inline tests/cpp_validation_tests/test_validation.cpp:446]:
```suggestion
            std::filesystem::path const& case_dir = dir_entry.path();
```

File: .clang-tidy [hunk 2]
```diff
@@ -8,7 +8,6 @@ boost-*,
 bugprone-*,
 -bugprone-easily-swappable-parameters,
 -bugprone-exception-escape,
--bugprone-macro-parentheses,
```
Comment (mgovers) [inline .clang-tidy:11]: cool! i guess this was resolved in the metadata refactoring by tony

Comment (mgovers) [reply .clang-tidy:11]: can you also check the other clang tidy issues listed in the aliases in https://clang.llvm.org/extra/clang-tidy/checks/list.html ?

**Review by mgovers — APPROVED** [review]

**Comment by Laurynas-Jagutis** [comment]
In the end, during the git hub checks, bugprone-macro-parentheses displayed several code smells, which I was not notified of when building in vscode. I reintroduced bugprone-macro-parentheses to the clang-tidy file for the sake of finishing this PR. Will pick bugprone-macro-parentheses up next week.

**Comment by mgovers** [comment]
> In the end, during the git hub checks, bugprone-macro-parentheses displayed several code smells, which I was not notified of when building in vscode. I reintroduced bugprone-macro-parentheses to the clang-tidy file for the sake of finishing this PR. Will pick bugprone-macro-parentheses up next week.

yeah that would be my go-to way as well. good thinking!

**Review by mgovers — APPROVED** [review]

**Pull request merged**